examples/01_basic_ping_bot/src/main.rs

```diff
@@ -59,54 +58,68 @@ impl From<TungsteniteError> for Error {
 }
 
 fn main() {
-    tokio::run(try_main().map_err(|why| {
-        println!("Error running shard: {:?}", why);
-    }).boxed().compat(TokioDefaultSpawner));
+    tokio::run(main_future().map_err(|reason| {
+        eprintln!("Error when running shard {:?}:", reason);
+    }).boxed().compat());
 }
 
-async fn try_main() -> Result<(), Error> {
+async fn main_future() -> Result<(), Error> {
     env_logger::init();
-
-    // Configure the client with your Discord bot token in the environment.
-    let token = env::var("DISCORD_TOKEN").expect("Expected a token in the environment");
-
-    // Create a new shard, specifying the token, the ID of the shard (0 of 1),
-    // and a handle to the event loop
-    let mut shard = await!(Shard::new(token, [0, 1]).compat())?;
-    println!("Connected shard");
-    let mut messages = shard.messages().compat();
+    let token = env::var("DISCORD_TOKEN").expect("No token was provided for the bot.");
+    // Create new shard.
+    let mut shard = await!(Shard::new(token, [0, 1]).compat()).unwrap();
+    let mut messages = shard.messages().expect("No shard messages found.").compat();
+    println!("Shard is connected.");
 
     loop {
-        // Loop over websocket messages.
         let result: Result<_, Error> = try {
-            println!("Getting message");
             let message = await!(messages.next())??;
-
-            // Parse the websocket message into a serenity GatewayEvent.
+            println!("Receiving shard message {:?}", message);
+            // Parse websocket event.
             let event = shard.parse(&message)?;
+            let process = shard.process(&event);
+            if let Ok(Some(action)) = process {
+                match action {
+                    Action::Identify => {
+                        println!("Identify Requested from Shard 0.");
+                        if let Err(why) = shard.identify() {
+                            println!("There was an error when identifiying: {:?}", why);
+                            break;
+                        }
+                    },
+                    Action::Autoreconnect => {
+                        println!("Shard 0 told us to autoreconnect.");
```

Comment:
Don't you need to get the new messages stream here, as in the error handling below?

Reply:
Yeah, thanks for reminding me of this, I have since added that assignment.
https://github.com/Texlo-Dev/serenity/commit/aa204805bdb55d1d2537d3be184f449e815032c2